adapters/mqtt-vertx-base/src/main/java/org/eclipse/hono/adapter/mqtt/ErrorSubscription.java

```diff
@@ -205,13 +205,11 @@ public Key getKey() {
     public String getErrorPublishTopic(final MqttContext context, final int errorCode) {
         Objects.requireNonNull(context);
 
-        final String contextTopicDeviceId = Optional.ofNullable(context.topic())
-                .map(ResourceIdentifier::getResourceId)
-                .orElse(null);
+        final String contextDeviceId = context.deviceId();
         final String contextEndpoint = Optional.ofNullable(context.topic())
                 .map(ResourceIdentifier::getEndpoint)
                 .orElse("");
-        return getErrorPublishTopic(contextEndpoint, contextTopicDeviceId, context.correlationId(), errorCode);
+        return getErrorPublishTopic(contextEndpoint, contextDeviceId, context.correlationId(), errorCode);
```

Comment:
Shouldn't a mapper also be able to change/override the original endpoint, e.g. forward a message published on an `event/...` topic to a `telemetry/...` target address?

Reply:
I was wondering the same.
It is neither mentioned as a possibility in the documentation, nor implemented, though. The endpoint (`event` or `telemetry`) is always kept as is:
https://github.com/eclipse/hono/blob/cedb10966db42f8473d1cb56d3daee88f3eede6d/adapters/mqtt-vertx-base/src/main/java/org/eclipse/hono/adapter/mqtt/impl/HttpBasedMessageMapping.java#L268-L269

Reply:
Currently, the mappedMessage does not contain the endpoint. I like the thought but this might be a bit overkill for this PR, no?

Reply:
We'll proceed in the same way as we do for making sure that the tenant cannot be changed, i.e. we postpone this to another PR for 1.12.0.